src/adapter/src/coord/command_handler.rs

```diff
@@ -683,6 +688,87 @@ impl Coordinator {
                 "CREATE SUBSOURCE statements",
             ))),
 
+            Statement::CreateMaterializedView(mut cmvs) => {
+                // (This won't be the same timestamp as the system table inserts, unfortunately.)
+                let mz_now = if cmvs
+                    .with_options
+                    .iter()
+                    .any(materialized_view_option_contains_temporal)
+                {
+                    let timeline_context =
+                        match self.validate_timeline_context(resolved_ids.0.clone()) {
+                            Ok(tc) => tc,
+                            Err(e) => return ctx.retire(Err(e)),
+                        };
+                    let timeline = match timeline_context {
+                        TimelineContext::TimelineDependent(timeline) => timeline,
+                        TimelineContext::TimestampDependent
+                        | TimelineContext::TimestampIndependent => {
+                            // We default to EpochMilliseconds, similarly to `determine_timestamp_for`.
+                            // Note that we didn't accurately decide whether we are TimestampDependent
+                            // or TimestampIndependent, because for this we'd need to also check whether
+                            // `query.contains_temporal()`, similarly to how `peek_stage_validate` does.
+                            // However, this doesn't matter here, as we are just going to default to
+                            // EpochMilliseconds in both cases.
+                            Timeline::EpochMilliseconds
+                        }
+                    };
```

Comment:
I've changed to `timeline_context.timeline().unwrap_or(Timeline::EpochMilliseconds)`.

> It may even be helpful to add a method to TimelineContext like timeline_or_default() that defaults to EpochMilliseconds. You could use that here and in determine_timestamp_for to prevent them from diverging.

The logic in `determine_timestamp_for` seems slightly different: It calls `get_timeline(timeline_context)`, which returns `None` for `TimestampIndependent`.

Should I still create `timeline_or_default()`, or it's not important if I would be the only caller?